lab5/app/src/main/java/com/app/lab5/HttpURLConnectionActivity.java

```diff
@@ -0,0 +1,243 @@
+                    httpConn.setConnectTimeout(10000);
+                    httpConn.setReadTimeout(10000);
+
+                    // Get input stream from web url connection.
+                    InputStream inputStream = httpConn.getInputStream();
+
+                    // Create input stream reader based on url connection input stream.
+                    isReader = new InputStreamReader(inputStream);
+
+                    // Create buffered reader.
+                    bufReader = new BufferedReader(isReader);
+
+                    // Read line of text from server response.
+                    String line = bufReader.readLine();
+
+                    // Loop while return line is not null.
+                    while(line != null)
+                    {
+                        // Append the text to string buffer.
+                        readTextBuf.append(line);
+
+                        // Continue to read text line.
+                        line = bufReader.readLine();
+                    }
+
+                    // Send message to main thread to update response text in TextView after read all.
+                    Message message = new Message();
+
+                    // Set message type.
+                    message.what = REQUEST_CODE_SHOW_RESPONSE_TEXT;
+
+                    // Create a bundle object.
+                    Bundle bundle = new Bundle();
+                    // Put response text in the bundle with the special key.
+                    bundle.putString(KEY_RESPONSE_TEXT, readTextBuf.toString());
+                    // Set bundle data in message.
+                    message.setData(bundle);
+                    // Send message to main thread Handler to process.
+                    uiUpdater.sendMessage(message);
+                }catch(MalformedURLException ex)
+                {
+                    Log.e(TAG_HTTP_URL_CONNECTION, ex.getMessage(), ex);
+                }catch(IOException ex)
+                {
+                    Log.e(TAG_HTTP_URL_CONNECTION, ex.getMessage(), ex);
+                }finally {
+                    try {
+                        if (bufReader != null) {
+                            bufReader.close();
+                            bufReader = null;
+                        }
+
+                        if (isReader != null) {
+                            isReader.close();
+                            isReader = null;
+                        }
+
+                        if (httpConn != null) {
+                            httpConn.disconnect();
+                            httpConn = null;
+                        }
+                    }catch (IOException ex)
+                    {
+                        Log.e(TAG_HTTP_URL_CONNECTION, ex.getMessage(), ex);
+                    }
+                }
+            }
+        };
+        // Start the child thread to request web page.
+        sendHttpRequestThread.start();
+    }
```

Comment:
coding style mix: please use Code -> reformat code.
Also, please remove all the unnecessary empty lines